markdown.dtx

```diff
@@ -25871,35 +26274,128 @@ M.extensions.tex_math_dollars = function()
         return (starter * C(p * (p - ender)^0) * ender)
       end
 
-      local inlinemathtail  = B( parsers.any * parsers.nonspacechar
-                               + parsers.backslash * parsers.any)
-                            * parsers.dollar 
-                            * -#(parsers.digit)
+      local allowed_before_closing = B( parsers.backslash * parsers.any
+                                      + parsers.any * (parsers.nonspacechar - parsers.backslash))
+
+      -- tex_math_dollars
```

Comment:
Instead of an opaque comment `-- tex_math_dollars`, how about a sentence or two in natural language?

```suggestion
%    \end{macrocode}
% \begin{markdown}
%
% The following patterns implement the Pandoc `tex_math_dollars` syntax extension.
%
% \end{markdown}
%  \begin{macrocode}
```